src/uniprotkb/components/entry/ComputationallyMappedSequences.tsx

```diff
@@ -0,0 +1,168 @@
+import AlignButton from '../../../shared/components/action-buttons/Align';
+import { Location, LocationToPath } from '../../../app/config/urls';
+
+// NOTE: Jie told me this would be replaced by a different
+// format he called "FASTA" entry
+type ProteinEntryLight = {
+  accession: string;
+  entryType: string;
+  sequenceLength: number;
+  geneName: string;
+  geneNameType: string;
+};
+
+type GeneCentricData = {
+  canonicalProtein: ProteinEntryLight;
+  relatedProteins: ProteinEntryLight[];
+};
+
+const ComputationalyMappedSequences: React.FC<{ primaryAccession: string }> = ({
+  primaryAccession,
+}) => {
+  const [selectedEntries, setSelectedEntries] = useState<string[]>([]);
+
+  const columns = [
+    {
+      label: 'Accession',
+      name: 'accession',
+      render: ({ accession, entryType }: ProteinEntryLight) => {
+        return (
+          <Link to={`/uniprotkb/${accession}`}>
+            <ReviewedUnreviewed
+              entryType={
+                entryType.match(/TrEMBL/i)
+                  ? EntryType.UNREVIEWED
+                  : EntryType.REVIEWED
+              }
+            />
+            {accession}
+          </Link>
+        );
+      },
+    },
+    {
+      label: 'Gene name',
+      name: 'gene_name',
+      render: ({ geneName }: ProteinEntryLight) => geneName,
+    },
+    {
+      label: 'Gene name type',
+      name: 'gene_name_type',
+      render: ({ geneNameType }: ProteinEntryLight) => geneNameType,
+    },
+    {
+      label: 'Length',
+      name: 'length',
+      render: ({ sequenceLength }: ProteinEntryLight) => sequenceLength,
+    },
+  ];
+
+  // Hooks
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const { data, loading, error, status } = useDataApi<GeneCentricData>(
+    apiUrls.genecentric(primaryAccession)
+  );
+
+  // Note: this function is duplicated in ResultsContainer.tsx
+  const handleSelectedEntries = useCallback((rowId: string) => {
+    setSelectedEntries((selectedEntries) => {
+      const filtered = selectedEntries.filter((id) => id !== rowId);
+      return filtered.length === selectedEntries.length
+        ? [...selectedEntries, rowId]
+        : filtered;
+    });
+  }, []);
+
+  const filteredData = useMemo(() => {
+    if (!data) {
+      return null;
+    }
+    return data.relatedProteins.filter(
+      ({ accession, geneNameType }) =>
+        geneNameType === 'Gene name' && !accession.startsWith(primaryAccession)
+    );
+  }, [primaryAccession, data]);
+
+  const handleViewAll = useCallback(() => {
+    const queryString = filteredData
+      ?.map(({ accession }) => `accession:${accession}`)
+      .join(' OR ');
+    history.push(
+      `${LocationToPath[Location.UniProtKBResults]}?query=(${queryString})`
+    );
+  }, [history, filteredData]);
+
+  if (error) {
+    if (status === 404) {
+      // Fail silently, this just means there's no data
+      return null;
+    }
+    dispatch(
```

Comment:
Not sure if this should pop-up a message here. This is just one section of an entry, and the user might not even need it or see it whenever they see the pop-up. I would have that as a localised error message inlined in the card

Reply:
Good point, I'd forgotten this was the approach we decided to take. Will update accordingly